### DISCUSSION TITLE
Connecting HomeSpan devices to other smarthome systems (e.g. homeassistant) possible?

### DISCUSSION THREAD

**Discussion by JLIspace** [HomeSpan/HomeSpan#752] [Q&A]
Hey,

I've made multiple rgb-lights, a standing-desk, fan, 433Mhz bridge, blind/shade controller and an IR-blaster, all while using HomeSpan. I really like the framework, but what it does lack is a standard way to inferface with none-HomeKit/apple users.
Does anyone know how I could (easily) connect all of my homespan devices to e.g. homeassistant (without using HA as a HomeKit controller) while keeping the normal HomeSpan HomeKit-functionality?

Thanks!

Julian

**Comment by jaycosmos** [comment]
I've been looking at Home Assistant(HA) for slightly different reasons. For sensors and collecting data, HA provides out of the box simple setup, data collection, and viewing options. Web based and on your phone. I don't care about automating my devices using HA on the other hand. All devices I've built need to be self sufficient. Relying on wifi or ethernet to be controlled is not an option.

I digress.... The answer is yes, but you'll need to add some way to communicate with HA. That is, you'll need to develop some layer of code that will bridge between HA and HomeSpan. See [Esp with home assistant WITHOUT esphome](https://community.home-assistant.io/t/esp-with-home-assistant-without-esphome/494713).

The HomeSpan API certainly gives you the flexibility to do this in your sketches, and then you'll need to pick which way to go to communicate with HA. I'll personally go the MQTT way as I have MQTT/ESP8266 projects deployed that I want to upgrade to ESP32 and make accessible via HomeKit controllers, primarily the Home App on my phone.

I see some references to people successfully using MQTT with HomeSpan but I've found no examples. I don't think it should be too difficult to get going. Then, it's just a matter of interfacing with HA using their MQTT integration. For that, there are plenty of examples.

**Comment by homonto** [comment]
I have created HomeKit devices using this library and to integrate with Home Assistant I added MQTT
Quite simple and much more functional than HomeKit
Yes, it will almost double the code but this way you can have devices being controlled by both: Apple and Home Assistant.
